### PR TITLE
[gpx] Implement GPX export for the tracks with timestamps

### PR DESCRIPTION
**Pull request by kirylkaveryn**
This PR adds timestamps to the track points when export to GPX is used.

### CODE REVIEW

File: kml/serdes_gpx.cpp [hunk 1]
```diff
@@ -492,15 +492,31 @@ void SaveTrackData(Writer & writer, TrackData const & trackData)
...
       auto const [lat, lon] = mercator::ToLatLon(point);
+      
       writer << kIndent4 << "<trkpt lat=\"" << CoordToString(lat) << "\" lon=\"" << CoordToString(lon) << "\">\n";
+      
       if (trackHasAltitude)
         writer << kIndent6 << "<ele>" << CoordToString(point.GetAltitude()) << "</ele>\n";
+
+      if (lineHasTimestamps)
+      {
+        CHECK_EQUAL(line.size(), timestampsForLine.size(), ());
```
Comment (biodranik) [inline kml/serdes_gpx.cpp:516]: This check should be after L500, not here in the loop:
```
if (lineHasTimestamps)
  CHECK_EQUAL(line.size(), timestampsForLine.size(), ());
```

**Review by biodranik — COMMENTED** [review]
Thanks! One comment.

Did you test it with different types of tracks, with and without timestamps?

**Comment by kirylkaveryn** [comment]
> Thanks! One comment.
> 
> 
> 
> Did you test it with different types of tracks, with and without timestamps?

We have a unit test that imports gpx, export the result data as a file and compare the results. 
I've added a new track with timestamps so the source file has 2 tracks: with and without timestamps. 
And all works well.

File: kml/serdes_gpx.cpp (hunk 1 above)
Comment (kirylkaveryn) [reply kml/serdes_gpx.cpp]: Fixed! Thanks

**Review by biodranik — APPROVED** [review]
Thanks!

**Pull request merged**